CORE-15757 WIP:some initial changes to setup the feature branch with avro objects to begin development

LWogan opened the pull request:
https://github.com/corda/platform-eng-design/pull/560

https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4552491009/Corda+Scalable+Topology+Design.

https://github.com/corda/corda-api/pull/1208

Theres a lot of logic i intend to change here. so please contact via slack rather than leave comments

Other WIP PRS:
Session init refactor + miscellaneous tickets: https://github.com/corda/corda-runtime-os/pull/4483
Session Close refactor: https://github.com/corda/corda-runtime-os/pull/4482

LWogan committed:
CORE-15757 some initial changes to setup the feature branch with avro objects to begin development

LWogan committed:
CORE-15757 some initial changes to setup the feature branch with avro objects to begin development

LWogan committed:
CORE-15757 disabled some tests

LWogan committed:
CORE-15757 detekt

LWogan committed:
CORE-15757 disabled test

chrisr3 commented:
This looks "extensive". Is the idea that it will improve the flow engine's performance? And if so, how quickly will you be able to confirm that this approach is successful?

LWogan commented:
> This looks "extensive". Is the idea that it will improve the flow engine's performance? And if so, how quickly will you be able to confirm that this approach is successful?


This work is part of https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4552491009/Corda+Scalable+Topology+Design. POCs have already confirmed that reducing the messages to only data messages was very successful at improving performance. Mainly through the removal of "pure ack" messages which reduces total session message count by half.

LWogan committed:
CORE-15757 updated api version

LWogan committed:
CORE-15757 compile error

LWogan committed:
CORE-15757 add require close to the flow session code. Last of initial code changes to feature to allow split of work

LWogan committed:
CORE-15757 disable e2e tests while dev is ongoing

LWogan committed:
CORE-15757 compile error

LWogan committed:
CORE-15757 fix som tests

LWogan committed:
CORE-16184 combined worker e2e test disabled

LWogan committed:
Merge remote-tracking branch 'origin/release/os/5.1' into feature/CORE-15757/flow-session-refactor

LWogan committed:
CORE-15757 update api version

LWogan committed:
Merge remote-tracking branch 'origin/release/os/5.1' into feature/CORE-15757/flow-session-refactor

# Conflicts:
#	gradle.properties

LWogan committed:
CORE-15757 merge from release

emilybowe committed:
CORE-15069 move flow to components (#4164)

Move two flow modules from libs to components. This will be used in future work to make use of a membership service in components to check clusters are local. Moving or copying this service was considered also but the move is also suggested for better design as these modules are closer to components than libs.

emilybowe committed:
CORE-15070: use service to check if cluster is local (#4201)

This PR adds a RecordFactory and slightly redesigns Init, Error and Event Executors. These executors now check if Outbound events are destined for the local cluster via the record factory and selects which topic to publish to. The RecordFactory uses a lookup service from membership to do this.

LWogan committed:
Merge remote-tracking branch 'origin/release/os/5.1' into feature/CORE-15757/flow-session-refactor

# Conflicts:
#	gradle.properties

LWogan committed:
CORE-15757 merge from release branch

LWogan committed:
CORE-16174 session init refactor, removal of heartbeating and resends (#4483)

SessionInit nullable field added to SessionData object. Populated for each data message sent to the initiated party until they respond with a data message. Initiator sessionState status will remain in CREATED until they receive a data message back (CONFIRMED)
We cannot be sure which SessionData will arrive first to the counterparty due to no order guarantees in the p2p layer. SessionInit object contains the information needed to start a flow on the other side, as well as the user and platform context data. This allows the checkpoint to be created regardless of which data message arrives first.

Removed restriction from initiating flow whereby they could not proceed past the first send without receiving something back.

contextSessionProperties is piggybacked onto existing data messages when it is not explicitly requested. It is moved to the SessionEvent object rather than the SessionConfirm object.

SessionInit & Confirm exchange is kept for when session.getCounterpartyInfo is called before any messages are exchanged.

Pure Acks removed

Heartbeating is removed

Session timeout logic left in place. Additional work required to integrate with the scheduler

Message resends logic removed.

A simple close mechanism implemented pending future work

LWogan committed:
CORE-16174 update api version

LWogan committed:
Merge remote-tracking branch 'origin/feature/CORE-15757/flow-session-refactor' into feature/CORE-15757/flow-session-refactor

# Conflicts:
#	gradle.properties

LWogan committed:
Merge remote-tracking branch 'origin/release/os/5.1' into feature/CORE-15757/flow-session-refactor

# Conflicts:
#	gradle.properties
